components/script/script_thread.rs

```diff
@@ -2453,10 +2724,14 @@ impl ScriptThread {
         }
     }
 
-    fn handle_resize(&self, id: PipelineId, size: WindowSizeData, size_type: WindowSizeType) {
+    /// Batch windo resize operations into a single "update the rendering" task,
+    /// or, if a load is in progress,
+    /// set the window size directly.
+    fn handle_resize_msg(&self, id: PipelineId, size: WindowSizeData, size_type: WindowSizeType) {
         let window = self.documents.borrow().find_window(id);
         if let Some(ref window) = window {
-            window.set_resize_event(size, size_type);
+            self.rendering_opportunity(id);
```

Comment:
This isn't in the specification right? It should probably have a todo along the lines of:

```rust
// TODO: This is not in the specification, but tests fail without explicitly triggering a rendering opportunity
// here. This should be removed.
```

Reply:
The spec doesn't specify when exactly a rendering opportunity should be noted.